task-1/src/Shape.ts

```diff
@@ -0,0 +1,37 @@
+import { Point } from './Point';
+
+export abstract class Shape {
+  protected color: string;
+  protected filled: boolean;
+  protected points: Point[];
+
+  constructor(points: Point[], color?: string, filled?: boolean);
+  constructor(points: Point[], color: string, filled: boolean) {
+    if (points.length < 3) {
+      throw new Error;
+    }
+
+    this.points = points;
+    this.color = color ?? "green";
+    this.filled = filled ?? true;
+  }
+
+  private isLastIndex = (index: number, arrayLength: number) => index === arrayLength - 1;
```

Comment:
This function is very general to be the private method of the abstract class. I would suggest remake it either as:
The util function
```ts
const isLastIndex = (index: number, arrayLength: number) => index === arrayLength - 1;
```
The protected isLastPoint method
```ts
protected isLastPoint(pointIndex: number): number { return pointIndex === this.points.length - 1 }
```
Or, I think the best one, to protected getNextPoint
```ts
protected getNextPoint(pointAIndex: number): Point {
  return pointAIndex === this.points.length - 1 ? this.points[0] : this.points[pointAIndex + 1];
}
```